example/transition_vtol_fixed_wing/transition_vtol_fixed_wing.cpp

```diff
@@ -0,0 +1,143 @@
+        std::cout << "Discovered device with UUID: " << uuid << std::endl;
+        discovered_device = true;
+    });
+
+    // We usually receive heartbeats at 1Hz, therefore we should find a device after around 2 seconds.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    if (!discovered_device) {
+        std::cout << ERROR_CONSOLE_TEXT << "No device found, exiting." << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+    // We don't need to specify the UUID if it's only one device anyway.
+    // If there were multiple, we could specify it with:
+    // dc.device(uint64_t uuid);
+    Device &device = dc.device();
+
+    // We want to listen to the altitude of the drone at 1 Hz.
+    const Telemetry::Result set_rate_result = dc.device().telemetry().set_rate_position(1.0);
+    if (set_rate_result != Telemetry::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Setting rate failed:" << Telemetry::result_str(
+                      set_rate_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+
+    // Set up callback to monitor altitude while the vehicle is in flight
+    device.telemetry().position_async([](Telemetry::Position position) {
+        std::cout << TELEMETRY_CONSOLE_TEXT // set to blue
+                  << "Altitude: " << position.relative_altitude_m << " m"
+                  << NORMAL_CONSOLE_TEXT // set to default color again
+                  << std::endl;
+    });
+
+
+    // Check if vehicle is ready to arm
+    if (device.telemetry().health_all_ok() != true) {
+        std::cout << ERROR_CONSOLE_TEXT << "Vehicle not ready to arm" << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+    // Arm vehicle
+    std::cout << "Arming..." << std::endl;
+    const Action::Result arm_result = device.action().arm();
+
+    if (arm_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Arming failed:" << Action::result_str(
+                      arm_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+    // Take off
+    std::cout << "Taking off..." << std::endl;
+    const Action::Result takeoff_result = device.action().takeoff();
+    if (takeoff_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Takeoff failed:" << Action::result_str(
+                      takeoff_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+	// Wait
+	std::this_thread::sleep_for(std::chrono::seconds(10));
+
+	// Change to FW
+    const Action::Result fw_result = device.action().transition_to_fixedwing();
+
+    if (fw_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Transition to fixed wing failed: " << Action::result_str(
+                      fw_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        //return 1;
+    }
+
+	// Wait
+	std::this_thread::sleep_for(std::chrono::seconds(10));
+
+    // Change to VTOL
```

Comment:
I'm not planning on fully reviewing this example just yet, but please add console output for vtol steps in the same way as we have for other (e.g. for taking off ... ` std::cout << "Taking off..." << std::endl;`). That makes the whole thing much more interesting when viewing operation.

When you have API to show vehicle state  you would probably use that to show transitions too.

Reply:
I agree.